draft-ietf-cose-hash-envelope.md

```diff
@@ -128,31 +129,41 @@ Profiles that rely on this specification MAY choose to mark TBD_1, TBD_2, TBD_3
 
 # Envelope EDN
 
-A hashed payload functions equivalently to an attached payload, with the benefits of being compact in size and providing the ability to validate the signature.
+The following informative example demonstrates how to construct a hash envelope for a resource already commonly referenced by its hash.
 
 ~~~~ cbor-diag
-18(                                    / COSE Sign 1               /
-   [
-    <<{
-      1:-35,                          / alg : ES384                /
-      4: h'75726e3a...32636573',      / kid                        /
-      16: "application/example+cose", / typ                        /
-      TBD_1: -16                      / payload_hash_alg : sha-256 /
-                                   / payload_preimage_content_type /
-      TBD_2: 51                    / "application/json-patch+json" /
-      TBD_3: "https://blob.example/a24f9c19"/ payload_location     /
-    }>>
-    {}                                 / Unprotected               /
-    h'935b5a91...e18a588a',            / Payload                   /
-    h'15280897...93ef39e5'             / Signature                 /
-   ]
-)
+18([ # cose-sign1
+  <<{
+    / signature algorithm / 1: -35, # ES384
+    / key identifier      / 4: h'75726e3a...32636573',
+    / cose sign1 type     / 16: "application/example+cose",
+    / hash algorithm      / TBD_1: -16, # sha256
+    / media type          / TBD_2: "application/spdx+json",
+    / location            / TBD_3: "https://sbom.example/" + ... + "/manifest.spdx.json"
+  }>>
+  / unprotected / {},
+  / payload     / h'935b5a91...e18a588a', # As seen in manifest.spdx.json.sha256
+  / signature   / h'15280897...93ef39e5'  # ECDSA Signature with SHA 384 and P-384
+])
 ~~~~
 
-In this example, the sha256 hash algorithm (-16) is used to hash the payload, which is of content type `application/json-patch+json` identified by the content format `51`.
-The full payload is located at "https://storage.example/244f...9c19".
-The COSE_sign1 is of type "application/example+cose".
-The sha256 hash is signed with ES384 which starts by taking the sha384 hash of the payload (which is a sha256 hash).
+In this example, an spdx software bill of materials (sbom) in json format is already commonly identified with its sha256 hash function, for example many tools will generate a file called `manifest.spdx.json.sha256` which contains that sha256 hash of the `manifest.spdx.json`.
+
+The content type for `manifest.spdx.json` is already well known as `application/spdx+json`, and is registered with IANA [here](https://www.iana.org/assignments/media-types/application/spdx+json).
+
+The full json software bill of material is available at the URL `https://sbom.example/.../manifest.spdx.json`.
+
+The payload of this cose-sign1 is the sha256 hash of the `manifest.spdx.json`, which is sometimes found in an adjacent file called `manifest.spdx.json.sha256`.
```

Comment:
```suggestion
The payload of such COSE_Sign1 are the SHA256 hashes of the content data in files, such as  `manifest.spdx.json`. Such COSE_Sign1 documents are typically located adjacent corresponding SHA256 hash files, e.g., `manifest.spdx.json.sha256`.
```

Tried to fix, but what? the payload of the COSE envelope is somtimes found in the hash file? Maybe I am reading this wrong.